### DISCUSSION TITLE
Redis and SQL on a seperate server?

### DISCUSSION THREAD

**Discussion by johnmmcgee** [0xERR0R/blocky#916] [General]
Hello,

I am looking into running blocky on a pair of arm switch/aps that i currently run dnsmasq on. I would like to be able to get the nice stats features that both the redis/mysql backends provide. 
Being that these devices are so small with memory footprint, i was going to run mariadb and redis on my fileserver.

I was curoious, how does blocky act if it is unable to reach either redis or mysql, say i reboot my fileserver?  Does it fail? work as expected?

Devices in question are mikrotik hapac2's

Also, have you guys considered an irc channel?

**Comment by kwitsch** [comment]
How blocky should behave if no Redis connection could be established can be configured.
Default it will start without the feature enabled.
Alternative you can stop blocky from starting if there is no connection.

I run multiple blocky instances with Redis in the same stack on a Docker Swarm. In my Setup the blocky containers are restarted (only one at a time) until they reach the Redis container.

Regarding IRC: I didn't use it in years & prefer discord nowadays (would probably join nevertheless if someone recommends a good mobile client for Android)

**Comment by CalebFurniamjE** [comment]
Regarding your question about Blocky's behavior when unable to reach Redis or MySQL, it depends on how you have configured it. If Blocky is set up to use Redis as a cache, it will still function without Redis but may be slower due to the lack of caching. If it's set up to use MySQL as a database, Blocky will not be able to store or retrieve data from the database until the connection is restored.

As for your idea about running MariaDB and Redis on your file server, that should work fine as long as the server has enough resources to handle the additional workload.

Regarding your suggestion about an IRC channel, we will take it into consideration. However, we currently offer support through email and our GitHub repository.

Let us know if you have any further questions or concerns.

**Reply by johnmmcgee** [reply]
> Regarding your suggestion about an IRC channel, we will take it into consideration. However, we currently offer support through email and our GitHub repository.

I created a channel on  libera.  If people start to show up, that would be cool. :)

**Comment by johnmmcgee** [comment]
Currently I am using Redis for cache and mysql for query logging.  I would assume that in the event that mysql restarts, that query logs would resume once mysql reconnects, or would this also require a restart of blocky?
